src/App.tsx

```diff
@@ -15,48 +16,116 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+// const SORT_FIELD_ALPHABET = 'alphabet';
+// const SORT_FIELD_LENGTH = 'length';
+
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  NoSort = 'no sort',
+}
+
+interface ObjectOfChanges {
+  sortType: SortType;
+  isReverse: boolean;
+}
+
+function getSortedGoods(
+  listOfGoods: string[],
+  { sortType, isReverse }: ObjectOfChanges,
+) {
+  const copyOfGoods = [...listOfGoods];
+
+  copyOfGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReverse === true) {
+    copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+}
+
+export const App = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NoSort);
+  const sortedGoods
```

Comment:
Don't forget about spaces
```suggestion

  const sortedGoods
```